arviz/stats/diagnostics.py

```diff
@@ -195,10 +199,12 @@ def ess(data, *, var_names=None, method="bulk", relative=False, prob=None):
 
     ufunc_kwargs = {"ravel": False}
     func_kwargs = {"relative": relative} if prob is None else {"prob": prob, "relative": relative}
-    return _wrap_xarray_ufunc(ess_func, dataset, ufunc_kwargs=ufunc_kwargs, func_kwargs=func_kwargs)
+    return _wrap_xarray_ufunc(
+        ess_func, dataset, ufunc_kwargs=ufunc_kwargs, func_kwargs=func_kwargs, **kwargs
+    )
 
 
-def rhat(data, *, var_names=None, method="rank"):
+def rhat(data, *, var_names=None, method="rank", **kwargs):
```

Comment:
Per your previous question I would not include `**kwargs` here. The reason is that previously the API was easy to introspect, but now anything could be in kwargs and the flexibility comes at the cost of readability. If this namspace is used solely for ask I would say either add the dask keyword args explicitly, or if the dask args are not known ahead of time change this to `dask_kwargs` so throughout the method is easy to see what is related to ask.

Let me know if this seems crazy, happy to take feedback!

Reply:
I would argue against adding Dask kwargs explicitly as it is still an area very actively developed in both xarray and dask. If you look at the [blame](https://github.com/pydata/xarray/blame/master/xarray/core/computation.py#L771) of dask related kwargs, you'll see the `meta` kwarg was added only 5 months ago for example. And yet, `apply_ufunc` with dask does not support multiple outputs with dask (I think) but will do so in the near future which could add even another dask related kwarg. 

Adding them explicitly would force us to keep up to date with xarray modifications (not only of new kwargs but also when defaults are modified).

Reply:
I think the ideal behaviour would be having the default `dask_kwargs` in `Dask` class but exposing `dask_kwargs` too to user functions. Using `dask_kwargs` instead of `**kwargs` I think is a good idea because even though these kwargs can theoretically be used to pass non dask parameters, it is a bad idea to pass non dask parameters. `dask_kwargs` helps getting this accross.

Regarding documentation (so things don't end up being too obscure) I think the ideal would be linking to `wrap_xarray_ufunc` or `xarray.apply_ufunc` and also to a guide about dask usage. The same way `xarray.DataArray.sel` has examples in API docs but still links to the Indexing section of the User Guide.

Reply:
Agree